src/lib.rs

```diff
@@ -858,6 +865,28 @@ impl Bump {
         }
     }
 
+    /// Similar to alloc_with, but rewrite the memory in place without any allocation.
+    #[inline(always)]
+    #[allow(clippy::mut_from_ref)]
+    pub fn alloc_with_inplace<F, T>(&self, place: NonNull<u8>, f: F) -> &mut T
```

Comment:
Neither of these functions can be safe because there is no guarantee that the `NonNull<u8>` is a valid memory location to store a `T`, let alone previously allocated by this `Bump`.